flowise-embed/src/components/Bot.tsx

```diff
@@ -357,128 +367,155 @@ export const Bot = (botProps: BotProps & { class?: string }) => {
     handleSubmit(prompt);
   };
 
-  // Handle form submission
-  const handleSubmit = async (value: string, action?: IAction | undefined | null) => {
-    setUserInput(value);
+  const handleSubmit = async (inputValue: string, action?: IAction | null) => {
+    try {
+      setUserInput(inputValue);
+      setLoading(true);
+      scrollToBottom();
+      clearPreviews();
 
-    if (value.trim() === '') {
-      const containsAudio = previews().filter((item) => item.type === 'audio').length > 0;
-      if (!(previews().length >= 1 && containsAudio)) {
-        return;
-      }
-    }
+      const fileUploads = getFileUploads();
 
-    setLoading(true);
-    setUploading(false);
-    scrollToBottom();
+      if (props.flow === Flow.CriticalAnalysis.toString()) {
+        const promptInformMissingData = `CORRIGI_JSON\n${JSON.stringify(jsonResponseCriticalAnalysis())}\ntext:${inputValue}`;
+        const jsonCriticalAnalysisUpdate = await sendBackgroundMessage(promptInformMissingData, fileUploads);
+        updateMessages(inputValue, fileUploads);
+        await processCriticalAnalysisUpdate(jsonCriticalAnalysisUpdate);
+      } else {
+        updateMessages(inputValue, fileUploads);
+        const body: IncomingInput = {
+          question: inputValue,
+          chatId: chatId(),
+        };
+        if (fileUploads && fileUploads.length > 0) body.uploads = fileUploads;
+        if (props.chatflowConfig) body.overrideConfig = props.chatflowConfig;
+        if (leadEmail()) body.leadEmail = leadEmail();
+        if (action) body.action = action;
+        if (isChatFlowAvailableToStream()) {
+          body.socketIOClientId = socketIOClientId();
+        } else {
+          setUploading(false);
+          setMessages((prevMessages) => [...prevMessages, { message: '', type: 'apiMessage' }]);
+        }
+        const result = await sendMessageQuery({
+          chatflowid: props.chatflowid,
+          apiHost: props.apiHost,
+          body,
+        });
+        if (result.data) {
+          const data = result.data;
+          let text = '';
+          if (data.text) text = data.text;
+          else if (data.json) text = JSON.stringify(data.json, null, 2);
+          else text = JSON.stringify(data, null, 2);
+          updateLastMessage(text, data?.sourceDocuments, data?.fileAnnotations, data?.agentReasoning, data?.action, data.text);
+        }
+        if (result.error) {
+          const error = result.error;
+          console.error(error);
+          if (typeof error === 'object') {
+            handleError(`Error: ${error?.message.replaceAll('Error:', ' ')}`);
+            return;
+          }
+          if (typeof error === 'string') {
+            handleError(error);
+            return;
+          }
+          handleError();
+          return;
```

Comment:
E se fizéssemos algo assim:
```suggestion
          let errorMessage = error.toString();
          if (typeof error === 'object') {
            errorMessage = `Error: ${error?.message.replaceAll('Error:', ' ')}`;
          }
          handleError(errorMessage);
          return;
```
?